crates/sovereign-sdk/module-system/sov-modules-stf-blueprint/src/lib.rs

```diff
@@ -196,34 +201,35 @@ where
         witness: <<C as Spec>::Storage as Storage>::Witness,
         slot_header: &<Da as DaSpec>::BlockHeader,
         soft_confirmation: &mut SignedSoftConfirmationBatch,
-    ) -> (Result<(), ApplySoftConfirmationError>, WorkingSet<C>) {
+    ) -> (Result<(), SoftConfirmationError>, WorkingSet<C>) {
         native_debug!("Applying soft confirmation in STF Blueprint");
 
+        let checkpoint = StateCheckpoint::with_witness(pre_state, witness);
```

Comment:
Declare those vars right before using them not before pub key checks

Reply:
used in return Err arm

https://github.com/chainwayxyz/citrea/issues/1047